overlord/snapshotstate/backend/backend.go

```diff
@@ -140,6 +144,17 @@ func Iter(ctx context.Context, f func(*Reader) error) error {
 			if !ok {
 				continue
 			}
+			// keep track of in-progress in a map as well
+			// to avoid races from the fact that we read only
```

Comment:
I'm not quite sure why this mentions "races", it's more related to the order on things on disk, no??

Reply:
Let me given an example. The Iter() function reads 100 dirnames and calls the callback for each of them. So there is a race if we have a snapshot import with 2+ snaps like:
1.dir-names names are read
2. 99_some-snap_1.0_x1.zip is there
3. code checks if 99_importing is there, it is so 99_some-snap is skipped
4. other snapshots are examined
5. in-parallel 99_importing finishes
7. 99_other-snap_1.0_x1.zip is there too
6. code checks if 99_importing is there, it is no longer there because import finished in the meantime but we still want to not call the callback with 99_other-snap or the callback would get an incomplete view that snapshot ID 99 consists of 2+ snapshots.

Does the above makes sense? If so I will add it into the comment.

Reply:
yes, that makes sense